src/query/query_opfunc.c

```diff
@@ -589,6 +585,16 @@ qdata_generate_tuple_desc_for_valptr_list (THREAD_ENTRY * thread_p, VALPTR_LIST
 	      status = QPROC_TPLDESCR_FAILURE;
 	      goto exit_with_status;
 	    }
+
+	  /* The value has been peeked so it does not require any clear, but we still might have the compressed_string
+	   * alloced so we need to clear it.
+	   */
+	  val_buffer = tuple_desc_p->f_valp[tuple_desc_p->f_cnt];
+	  if (!DB_IS_NULL (val_buffer))
```

Comment:
You can add a variable and modify line 575:

```C
if (pr_is_set_type (DB_VALUE_DOMAIN_TYPE (tuple_desc_p->f_valp[tuple_desc_p->f_cnt])))
```
->
```C
dbval_type = DB_VALUE_DOMAIN_TYPE (tuple_desc_p->f_valp[tuple_desc_p->f_cnt]);
if (pr_is_set_type (dbval_type))
  {
    ...
  }
```

You can also rewrite line 593 as follows:
```C
  if (!DB_IS_NULL (val_buffer) && (dbval_type == DB_TYPE_VARCHAR || dbval_type == ...))
```